cmd/fetch_k8s_resource.go

```diff
@@ -46,90 +38,57 @@ func k8sResourceAuthorizationCmd() *cobra.Command {
+  ec fetch k8s-resource --policy "$(cat /path/to/ecp.json)"`,
 
-  ec fetch k8s-resource --file-path <path/to/ApplicationSnapshot/file> \
-      --public-key <path/to/public/key> --namespace <namespace> --resource <resource>
-
-Validate attestation of multiple images from an inline ApplicationSnapshot
-Spec and fetch authorizations:
-
-  ec fetch k8s-resource --json-input '{"components":[{"containerImage":"<image url>"}]}' \
-      --public-key <path/to/public/key> --namespace <namespace> --resource <resource>
-
-Use public key from a kubernetes secret:
-
-  ec fetch k8s-resource --image-ref <image url> \
-     --public-key k8s://<namespace>/<secret-name> --namespace <namespace> --resource <resource>`,
-		PreRunE: func(cmd *cobra.Command, args []string) error {
-			spec, err := applicationsnapshot.DetermineInputSpec(data.filePath, data.input, data.imageRef)
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := validateK8sSource(
+				cmd.Context(),
+				data.policyConfiguration,
+			)
 			if err != nil {
-				return err
+				log.Println(err)
 			}
 
-			data.spec = spec
-
-			return nil
-		},
-		RunE: func(cmd *cobra.Command, args []string) error {
-			for _, comp := range data.spec.Components {
-				err := validateK8sSource(
-					cmd.Context(),
-					comp.ContainerImage,
-					data.publicKey,
-					data.namespace,
-					data.server,
-					data.resource,
-				)
-				if err != nil {
-					log.Println(err)
-					continue
-				}
-			}
 			return nil
 		},
 	}
 
-	cmd.Flags().StringVar(&data.publicKey, "public-key", data.publicKey, "path to the public key associated with the image attestation")
-	cmd.Flags().StringVar(&data.imageRef, "image-ref", data.imageRef, "OCI reference to the attested image")
-	cmd.Flags().StringVarP(&data.filePath, "file-path", "f", data.filePath, "path to ApplicationSnapshot Spec JSON file")
-	cmd.Flags().StringVarP(&data.input, "json-input", "j", data.input, "JSON representation of an ApplicationSnapshot Spec")
-	cmd.Flags().StringVarP(&data.namespace, "namespace", "n", data.namespace, "namespace containing the EnterpriseContractPolicy resource")
-	cmd.Flags().StringVarP(&data.server, "server", "s", data.server, "kubernetes server URL containing the EnterpriseContractPolicy resource")
-	cmd.Flags().StringVarP(&data.resource, "resource", "r", data.resource, "name of the EnterpriseContractPolicy resource")
+	cmd.Flags().StringVarP(&data.policyConfiguration, "policy", "p", data.policyConfiguration,
+		"EntepriseContractPolicy reference [<namespace>/]<name>")
 
 	return cmd
 }
 
-func validateK8sSource(ctx context.Context, imageRef, publicKey, namespace, server, resource string) error {
-	k8sSource, err := image.NewK8sSource(namespace, server, resource)
+func GetK8sAuthorization(ecp *ecc.EnterpriseContractPolicySpec) error {
+	resource, err := image.GetK8sResource(ecp)
 	if err != nil {
 		return err
 	}
 
-	authorization, err := image.GetAuthorization(ctx, k8sSource)
+	auth, err := resource.GetSignOff()
```

Comment:
We should probably handle the error here
```suggestion
	auth, err := resource.GetSignOff()
	if err != nil {
		return err
	}
```